app/components/ImageImport/ImageImport.js

```diff
@@ -26,22 +26,13 @@ export default class ImportImage extends Component {
    let reader = new FileReader();
    let file = e.target.files[0];
 
-   let imagePromise = new Promise((resolve, reject)=>{
-     reader.readAsDataURL(file)
-     reader.onloadend = () => {
-       this.setState({
-         file: file,
-         imagePreviewUrl: reader.result
-       });
-     }
-     setTimeout(function(){
-       resolve("");
-     }, 200);
-   });
-   imagePromise.then(()=>{
-     console.log("Image conversion promise resolved.")
-     console.log(this.state)
-   })
+   reader.readAsDataURL(file)
+   reader.onloadend = () => {
+      this.setState({
```

Comment:
I like breaking this out of the image component, it looks a lot better and the onloadend was essentially doing the same as the timeout, right? Looking good! 

You might consider changing your console log errors on line 58 of App into something the user will see?